### PR TITLE
Fix: avoid accidental load of unnecessary Xcode setting lines

### PR DESCRIPTION
**Pull request by shoamano83**
The original implementation grep-ed "CONFIGURATION", therefore it captured any lines that contain the word, not just the line starting with `CONFIGURATION=`. This PR fixes the issue so that relax now captures `CONFIGURATION=` line only.

Background: we faced a weird build issue where Swift compilation failed after I added
```
OTHER_SWIFT_FLAGS = "$(inherited) -D CONFIGURATION_XXX";
```
line in project.pbxproj file. Looking at the code, I noticed that relax captured this setting and exported it, which caused Xcode to use wrong `OTHER_SWIFT_FLAGS` option.

### CODE REVIEW

File: libexec/util-build [hunk 1]
```diff
@@ -332,16 +332,16 @@ __load_xcode_build_settings () {
 		s/$/'/;
 		" | \
 		grep \
-		-e "CONFIGURATION\|INFOPLIST_PATH\|PRODUCT_SETTINGS_PATH\|SRCROOT"\
```
Comment (shoamano83) [inline libexec/util-build:335]: Let me double check the intention of original implementation ... was this to capture the line `CONFIGURATION=` only, or was this meant to capture other lines such as `CONFIGURATION_BUILD_DIR=` and `PODS_CONFIGURATION_BUILD_DIR=` ?

Comment (scenee) [reply libexec/util-build:335]: Yes it was,  `relax` uses only `CONFIGURATION` environment variable, not other variables including the word 👍

**Comment by scenee** [comment]
Thank you so much. This PR makes sense. Unfortunately, CI jobs are failed because they are too old to work on the current GitHub Actions environment. However it should work correctly after this change and then I merged this into the master branch.

File: libexec/util-build (hunk 1 above)
Comment (shoamano83) [reply libexec/util-build:335]: OK, and thank you!

**Pull request merged**